go-controller/pkg/ovn/master.go

```diff
@@ -962,15 +1014,27 @@ func (oc *Controller) allocateNodeSubnets(node *kapi.Node) ([]*net.IPNet, []*net
 		if allocatedHostSubnet != nil {
 			klog.V(5).Infof("Allocating subnet %v on node %s", allocatedHostSubnet, node.Name)
 			allocatedSubnets = append(allocatedSubnets, allocatedHostSubnet)
+			// Release the allocation on error
+			defer func() {
+				if err != nil {
+					klog.Warningf("Releasing subnet %v on node %s: %v", allocatedHostSubnet, node.Name, err)
+					errR := oc.masterSubnetAllocator.ReleaseNetwork(allocatedHostSubnet)
+					if errR != nil {
+						klog.Warningf("Error releasing subnet %v on node %s", allocatedHostSubnet, node.Name)
+					}
+				}
+			}()
+
 		}
 	}
 	// check if we were able to allocate the new subnets require
 	// this can only happen if OVN is not configured correctly
 	// so it will require a reconfiguration and restart.
 	wantedSubnets := expectedHostSubnets - currentHostSubnets
 	if wantedSubnets > 0 && len(allocatedSubnets) != wantedSubnets {
-		return nil, nil, fmt.Errorf("error allocating networks for node %s: %d subnets expected only new %d subnets allocated",
+		err = fmt.Errorf("error allocating networks for node %s: %d subnets expected only new %d subnets allocated",
```

Comment:
is this an accident?

Reply:
No, the release won't be called since the deferred function is called only when `err != nil` and since it's not a named return, I think we will leak if we hit this if conditional